examples/save_plot/src/main.rs

```diff
@@ -32,8 +32,12 @@ impl eframe::App for MyApp {
 
             // let's create a dummy line in the plot
             let graph: Vec<[f64; 2]> = vec![[0.0, 1.0], [2.0, 3.0], [3.0, 2.0]];
-            let inner = my_plot.show(ui, |plot_ui| {
+            let inner = my_plot.show(ui, |plot_ui_builder| {
+                let mut plot_ui = plot_ui_builder.into_plot_ui();
+
                 plot_ui.line(Line::new(PlotPoints::from(graph)).name("curve"));
+
+                ((), plot_ui)
```

Comment:
Every example now starts with `let mut plot_ui = plot_ui_builder.into_plot_ui();` and ends with `((), plot_ui)`.
This is a bit ugly.

I assume this is needed in order to get the lifetime of `plot_ui` correct?

Is there no way around it, e.g. assign a lifetime to `Plot` and use that for `plot_ui` too?

Reply:
Agreed, I also found this ugly.

Hmm, maybe.

I thought the root cause was the creation of PlotUi in side the show function which leads to lifetime issues with the build_fn.

What I also tried as a solution was have the return is an `impl Into<BuildFnResultThingy>` which is implemented for tuples of (R, PlotUi) and PlotUi etc. to make returning more easy but this also adds indirection as what you can return or not.